examples/srl/main_train_srl.py

```diff
@@ -0,0 +1,207 @@
+        LabeledSpanGraphNetwork:
+    text_extractor: BaseExtractor = schemes["text_tag"]["extractor"]
+    char_extractor: BaseExtractor = schemes["char_tag"]["extractor"]
+    link_extractor: BaseExtractor = schemes["pred_link_tag"]["extractor"]
+
+    _model = LabeledSpanGraphNetwork(word_vocab=text_extractor.get_dict(),
+                                     char_vocab_size=char_extractor.size(),
+                                     label_vocab=link_extractor.get_dict())
+
+    return _model
+
+
+def train(_model: LabeledSpanGraphNetwork,
+          _optim: Optimizer,
+          _batch: Batch) -> \
+        LabeledSpanGraphNetwork.ReturnType:
+    char_tensor: Tensor = _batch["char_tag"]["data"]
+    char_masks: List[Tensor] = _batch["char_tag"]["masks"]
+    text_tensor: Tensor = _batch["text_tag"]["data"]
+    text_mask: Tensor = _batch["text_tag"]["masks"][0]
+    text: List[List[str]] = _batch["raw_text_tag"]["data"]
+    pred_link_features: List[Feature] = _batch["pred_link_tag"]["features"]
+
+    _optim.zero_grad()
+
+    output: LabeledSpanGraphNetwork.ReturnType = \
+        model(text=text,
+              char_batch=char_tensor,
+              char_masks=char_masks,
+              text_batch=text_tensor,
+              text_mask=text_mask,
+              srl_features=pred_link_features)
+
+    output["loss"].backward()
+    optim.step()
+
+    return output
+
+
+def predict_forward_fn(_model: LabeledSpanGraphNetwork, _batch: Dict) -> Dict:
+    char_tensor: Tensor = _batch["char_tag"]["data"]
+    char_masks: List[Tensor] = _batch["char_tag"]["masks"]
+    text_tensor: Tensor = _batch["text_tag"]["data"]
+    text_mask: Tensor = _batch["text_tag"]["masks"][0]
+    text: List[List[str]] = _batch["raw_text_tag"]["data"]
+
+    # TODO: test enable enforce_constriant
+    model_output: List[Dict[int, List[data.Span]]] = \
+        _model.decode(text=text,
+                      char_batch=char_tensor,
+                      char_masks=char_masks,
+                      text_batch=text_tensor,
+                      text_mask=text_mask)
+
+    output: List[Dict] = []
+    for model_output_i in model_output:
+        # TODO: use extractor specified name
+        output_i: Dict = {
+            "data": [],
+            "parent_unit_span": [],
+            "child_unit_span": []
+        }
+        spans: List[data.Span]
+        for predicate_id, spans in model_output_i.items():
+            for span in spans:
+                output_i["data"].append(span.label)
+                output_i["parent_unit_span"].append(
+                    (predicate_id, predicate_id + 1))
+                output_i['child_unit_span'].append((span.start, span.end))
+
+    return {'pred_link_tag': output}
+
+
+if __name__ == "__main__":
+    num_epochs = 5
+    lr = 0.01
+    momentum = 0.9
+    nesterov = True
+    batch_size = 64
+
+    train_path = "data/train_tiny/"
+
+    tp_request: Dict = {
```

Comment:
How about we put the request in a config file to make this example main body more readable?